### PR TITLE
ENG-18423 DemoApp uses new Component + Hilt (Dagger)

### PR DESCRIPTION
**Pull request by daniftodi**
Ticket: https://extole.atlassian.net/browse/ENG-18423

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -1,78 +1,83 @@
...
-    id 'kotlin-android'
-    id "io.gitlab.arturbosch.detekt" version "1.17.1"
-    id "com.github.evestera.depsize" version "0.2.0"
+  id 'com.android.application'
+  id 'kotlin-android'
+  id "io.gitlab.arturbosch.detekt" version "1.17.1"
+  id "com.github.evestera.depsize" version "0.2.0"
+  id 'kotlin-kapt'
+  id 'com.google.dagger.hilt.android'
 }
```
Comment (mcyster) [inline app/build.gradle:8]: I hate it but we indent 4 spaces

File: app/src/main/java/com/extole/androidsdk/MainActivity.kt [hunk 2]
```diff
@@ -1,48 +1,41 @@
...
             runOnUiThread {
-                findViewById<EditText>(R.id.email_message).setText(
-                    zone.get("sharing.email.message").toString(), TextView.BufferType.NORMAL
-                )
-
-                findViewById<TextView>(R.id.copyright).setText(
-                    zone.get("sharing.facebook.description").toString(),
-                    TextView.BufferType.NORMAL
+                findViewById<Button>(R.id.menu_item).setText(
+                    zone.get("cta.title").toString(), TextView.BufferType.NORMAL
```
Comment (mcyster) [inline app/src/main/java/com/extole/androidsdk/MainActivity.kt]: cta_prefetch zone should probably not have "cta" repeated inside - re "cta.title" vs "title"?

Comment (mcyster) [reply app/src/main/java/com/extole/androidsdk/MainActivity.kt]: same below

File: app/src/main/java/com/extole/androidsdk/MainActivity.kt [hunk 3]
```diff
@@ -1,48 +1,41 @@
...
             }
 
-            launch {
-                val customMenu1 = campaign.getZone("custom_menu_1")
-                runOnUiThread {
-                    findViewById<Button>(R.id.menu_item).text =
-                        customMenu1.get("message").toString()
+            findViewById<Button>(R.id.deeplink).setOnClickListener {
+                GlobalScope.launch {
+                    extole.sendEvent("deeplink", mapOf("extole_key" to "extole_value"))
```
Comment (mcyster) [inline app/src/main/java/com/extole/androidsdk/MainActivity.kt:43]: can we have this pass extole_.*

**Review by mcyster — APPROVED** [review]
:+1:

can you test it against Alex P's mobile component?

File: app/src/main/java/com/extole/androidsdk/MainActivity.kt (hunk 3 above)
Comment (daniftodi) [reply app/src/main/java/com/extole/androidsdk/MainActivity.kt:43]: I don't think we need to pass it here as extole_.*, it is configured to trigger on `extole.*_` see https://mobile-monitor.extole.io/zone/mobile_bootstrap 

```
{
   "conditions":[
      {
         "type":"EVENT",
         "has_data_keys":[
            "extole_.*"
         ],
         "event_names":[
            "deeplink"
         ]
      }
   ],
   "actions":[
      {
         "type":"VIEW_FULLSCREEN",
         "zone_name":"microsite"
      }
   ]
}
```

File: app/src/main/java/com/extole/androidsdk/MainActivity.kt (hunk 2 above)
Comment (daniftodi) [reply app/src/main/java/com/extole/androidsdk/MainActivity.kt]: @apaniciuc let's change the zone: https://mobile-monitor.extole.io/zone/cta_prefetch response to:

```
{
   "title":"CTA Item",
   "image":"https://origin.xtlo.net/type=creativeArchive:clientId=1106500674:creativeArchiveId=7153594407382986000:version=1:coreAssetsVersion=66/images/generic-social.jpg",
   "touch_event":"cta_tap"
}
```

Comment (apaniciuc) [reply app/src/main/java/com/extole/androidsdk/MainActivity.kt]: I have changed this to
```
var mobileCta = {
    title: staticVariables.titleCopy,
    image: staticVariables.socialShareImageUrl,
    touch_event: "cta_tap"
};
```

**Pull request merged**